### DISCUSSION TITLE
Just a quick question from a curious developer

### DISCUSSION THREAD

**Discussion by maxacarvalho** [directus/directus#3321] [Archive]
Hello,

Please forgive me if that's not the place for such question but I'm cracking my mind and I can't figure this out.  
I also would like to say that your project is amazing!!!

Anyways, let me go for my question. I'm very much searching for nice components implementations and I came across you project. I was checking the `v-menu` approach and it's extremely interesting and flexible but, there's on piece that I can't get.

On the line https://github.com/directus/directus/blob/main/app/src/components/v-menu/v-menu.vue#L23 you have the `v-if="isActive"` for the menu element inside the portal.  

Then, you rely on the `popper.value = document.getElementById(id.value);` to setup PopperJs, **but**, as far as I see, when you run that call, due to the `v-if`, there's no element inside the portal, so how come the `document.getElementById` works?  
According to what I understand it will return always `null`, and if so, PopperJs will never start and the element position will never get right no? Is there anything that I'm missing?

Feel free to ignore my question if answering that makes no sense to you, and keep going with the incredible job!

**Comment by rijkvanzanten** [comment]
Heya!

Thanks for your kind words. 

> Please forgive me if that's not the place for such question but I'm cracking my mind and I can't figure this out.

I don't usually get these kinds of questions, that's for sure 😁 

> Then, you rely on the popper.value = document.getElementById(id.value); to setup PopperJs, but, as far as I see, when you run that call, due to the v-if, there's no element inside the portal, so how come the document.getElementById works?
According to what I understand it will return always null, and if so, PopperJs will never start and the element position will never get right no? Is there anything that I'm missing?

Note the context where we're doing this assignment:

https://github.com/directus/directus/blob/4e58751e71497768c3118b4d80666fe06f76f579/app/src/components/v-menu/v-menu.vue#L125-L134

First of all, we only do the assignment if active == true, so therefore only when the element is being rendered. You're right in your assessment that "there's no element inside the portal". However, that's where this comes in:

```js
Vue.nextTick(() => {});
```

With nextTick, we wait for the next update in Vue to have happened before executing the callback. This means that we're waiting for the element to be rendered before reading `document.getElementById(id.value)`, which means that in that point in time, it actually does exist!

**Reply by maxacarvalho** [reply]
Hi @rijkvanzanten, and thank you very much for answering my silly question.  
I completely missed that to the `Vue.nextTick(() => {});`!!!

Once again, sorry to bother ;-)

**Reply by rijkvanzanten** [reply]
No problem! Not a silly question at all 👍
